text/0497-glue-l2-construct.md

```diff
@@ -0,0 +1,434 @@
+    className: 'com.example.HelloWorld',
+});
+```
+
+### Workflow Triggers
+
+In AWS Glue, developers can use workflows to create and visualize complex
+extract, transform, and load (ETL) activities involving multiple crawlers,
+jobs, and triggers. Standalone triggers are an anti-pattern, so we will
+only create triggers from within a workflow.
+
+Within the workflow object, there will be functions to create different
+types of triggers with actions and predicates. Those triggers can then be
+added to jobs.
+
+For all trigger types, the StartOnCreation property will be set to true by
+default, but developers will have the option to override it.
+
+1. **On Demand Triggers**
+
+On demand triggers can start glue jobs or crawlers. We’ll add convenience
+functions to create on-demand crawler or job triggers. The trigger method
+will take an optional description but abstract the requirement of an actions
+list using the job or crawler objects using conditional types.
+
+```
+myWorkflow = new glue.Workflow(this, "GlueWorkflow", {
+    name: "MyWorkflow";
+    description: "New Workflow";
+    properties: {'key', 'value'};
+});
+
+myWorkflow.createOnDemandTrigger(this, 'TriggerJobOnDemand', {
+    description: 'On demand run for ' + myGlueJob.name,
+    actions: [glueJob1, glueJob2, glueJob3, glueCrawler]
+});
+```
+
+1. **Scheduled Triggers**
+
+Schedule triggers are a way for developers to create jobs using cron
+expressions. We’ll provide daily, weekly, and monthly convenience functions,
+as well as a custom function that will allow developers to create their own
+custom timing using the [existing event Schedule object](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_events.Schedule.html)
+without having to build their own cron expressions. (The L2 will extract
+the expression that Glue requires from the Schedule object). The trigger method will
+take an optional description and list of Actions which can refer to Jobs or
+crawlers via conditional types.
+
+```
+// Create Daily Schedule at 00 UTC
+myWorkflow.createDailyScheduleTrigger(this, 'TriggerCrawlerOnDailySchedule', {
+    description: 'Scheduled run for ' + myGlueJob.name,
+    actions: [ myGlueCrawler]
+});
+
+// Create Weekly schedule at 00 UTC on Sunday
+myWorkflow.createWeeklyScheduleTrigger(this, 'TriggerJobOnWeeklySchedule', {
+    description: 'Scheduled run for ' + myGlueJob.name,
+    actions: [glueJob1, glueJob2, glueJob3, glueCrawler]
+});
+
+// Create Custom schedule, e.g. Monthly on the 7th day at 15:30 UTC
+myWorkflow.createCustomScheduleJobTrigger(this, 'TriggerCrawlerOnCustomSchedule', {
+    description: 'Scheduled run for ' + myGlueJob.name,
+    actions: [glueJob1, glueJob2, glueJob3, glueCrawler]
+    schedule: events.Schedule.cron(day: '7', hour: '15', minute: '30')
+});
+```
+
+#### **3. Notify  Event Triggers**
+
+This type of trigger is only supported with Glue workflows. There are two types
```

Comment:
> This type of trigger is only supported with Glue workflows.

As opposed to what? All examples here show triggers being used with Glue workflows.

Reply:
Technically you can create a Trigger outside of a Glue workflow in the console, CLi, or using L1s, but the L2 will not allow you to do so because it's an anti-pattern. We can remove this or add this construct if it causes confusion.

Reply:
I think you should rephrase this to say something along the lines "workflows are mandatory".

Reply:
They are for the L2 but not for the Glue service itself. I'll add some language to clarify.